pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -117,30 +119,60 @@ func (r *Reconciler) create() error {
 		Items: metadataItems,
 	}
 
-	operation, err := r.computeService.InstancesInsert(r.projectID, zone, instance)
+	return instance, nil
+}
+
+// Create creates machine if and only if machine exists, handled by cluster-api
+func (r *Reconciler) create() error {
+	instance, err := r.newInstanceFromMachineScope()
 	if err != nil {
-		if reconcileWithCloudError := r.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{
-			Type:    v1beta1.MachineCreated,
-			Reason:  machineCreationFailedReason,
-			Message: err.Error(),
-			Status:  corev1.ConditionFalse,
-		}); reconcileWithCloudError != nil {
-			klog.Errorf("Failed to reconcile machine with cloud state: %v", reconcileWithCloudError)
-		}
-		return fmt.Errorf("failed to create instance via compute service: %v", err)
+		return err
 	}
-	if op, err := r.waitUntilOperationCompleted(zone, operation.Name); err != nil {
-		if reconcileWithCloudError := r.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{
-			Type:    v1beta1.MachineCreated,
-			Reason:  machineCreationFailedReason,
-			Message: err.Error(),
-			Status:  corev1.ConditionFalse,
-		}); reconcileWithCloudError != nil {
-			klog.Errorf("Failed to reconcile machine with cloud state: %v", reconcileWithCloudError)
-		}
-		return fmt.Errorf("failed to wait for create operation via compute service. Operation status: %v. Error: %v", op, err)
+
+	// Passing machine.UID makes this request and subsequent
+	// requests idempotent as long as _all_ the other parameters
+	// to InstancesInsert() are equal. Note: you cannot use the
+	// same UID for any other operation as it is now keyed to the
+	// insert operation.
+	operation, err := r.computeService.InstancesInsert(string(r.machine.UID), r.projectID, r.providerSpec.Zone, instance)
```

Comment:
Just a follow-on from our discussion from earlier:  This would make the create event immutable in this actuator, which is currently different from what we do in AWS (aws has code to remove stopped instances, apparently).  I'm in favor of the immutable approach as done here.